source/base/data_out_base.cc

```diff
@@ -4068,330 +4068,364 @@ namespace DataOutBase
   }
 
 
-
-  template <int dim, int spacedim>
-  void
-  write_povray(
-    const std::vector<Patch<dim, spacedim>> &patches,
-    const std::vector<std::string> &         data_names,
-    const std::vector<
-      std::tuple<unsigned int,
-                 unsigned int,
-                 std::string,
-                 DataComponentInterpretation::DataComponentInterpretation>> &,
-    const PovrayFlags &flags,
-    std::ostream &     out)
+  namespace
   {
-    AssertThrow(out, ExcIO());
+    template <int dim, int spacedim>
+    void
+    do_write_povray(const std::vector<Patch<dim, spacedim>> &,
+                    const std::vector<std::string> &,
+                    const PovrayFlags &,
+                    std::ostream &)
+    {
+      Assert(false,
+             ExcMessage("Writing files in POVRAY format is only supported "
+                        "for two-dimensional meshes."));
+    }
+
+
+
+    void
+    do_write_povray(const std::vector<Patch<2, 2>> &patches,
+                    const std::vector<std::string> &data_names,
+                    const PovrayFlags &             flags,
+                    std::ostream &                  out)
+    {
+      AssertThrow(out, ExcIO());
 
 #ifndef DEAL_II_WITH_MPI
-    // verify that there are indeed patches to be written out. most
-    // of the times, people just forget to call build_patches when there
-    // are no patches, so a warning is in order. that said, the
-    // assertion is disabled if we support MPI since then it can
-    // happen that on the coarsest mesh, a processor simply has no cells it
-    // actually owns, and in that case it is legit if there are no patches
-    Assert(patches.size() > 0, ExcNoPatches());
+      // verify that there are indeed patches to be written out. most
+      // of the times, people just forget to call build_patches when there
+      // are no patches, so a warning is in order. that said, the
+      // assertion is disabled if we support MPI since then it can
+      // happen that on the coarsest mesh, a processor simply has no cells it
+      // actually owns, and in that case it is legit if there are no patches
+      Assert(patches.size() > 0, ExcNoPatches());
 #else
-    if (patches.size() == 0)
-      return;
+      if (patches.size() == 0)
+        return;
 #endif
-    Assert(dim == 2,
-           ExcNotImplemented()); // only for 2-D surfaces on a 2-D plane
-    Assert(spacedim == 2, ExcNotImplemented());
+      constexpr int dim      = 2;
```

Comment:
CI complains that this not used

Reply:
That's aggravating because it is used below in line 4175 :-( I'm pushing a patch for it.

Reply:
Yes, it is used but only in debug mode.